### PR TITLE
[cleanup] asana pipeline

### PR DESCRIPTION
**Pull request by sh-rp**
Example cleanup of a pipeline in our repo

### CODE REVIEW

File: pipelines/asana_dlt/__init__.py [hunk 1]
```diff
@@ -34,48 +49,26 @@ def workspaces(access_token: str = dlt.secrets.value) -> t.Iterator[dict]:
...
+    Fetches and returns a list of projects for a given workspace from Asana.
+
+    Args:
+        workspace (dict): The workspace data.
+        access_token (str): The access token to authenticate the Asana API client, provided in the secrets file
+
+    Returns:
+        list[dict]: The project data for the given workspace.
+    """
     print(f"Fetching projects for workspace {workspace['name']}...")
```
Comment (rudolfix) [inline pipelines/asana_dlt/__init__.py:65]: we can remove those prints. we have `progress` argument to pipeline that does exactly that

File: pipelines/asana_dlt/__init__.py [hunk 2]
```diff
@@ -34,48 +49,26 @@ def workspaces(access_token: str = dlt.secrets.value) -> t.Iterator[dict]:
     write_disposition="replace",
 )
 @dlt.defer
-def projects(workspace, access_token: str = dlt.secrets.value) -> t.Iterator[dict]:
-    """Returns a list of projects for a given workspace"""
+def projects(
+    workspace: Dict[str, Any], access_token: str = dlt.secrets.value
+) -> Iterable[Dict[str, Any]]:
```
Comment (rudolfix) [inline pipelines/asana_dlt/__init__.py:54]: there are two types to mark returned items `TDataItems` in this case fits

File: pipelines/asana_dlt/__init__.py [hunk 3]
```diff
@@ -286,28 +240,47 @@ def teams(workspace, access_token: str = dlt.secrets.value) -> t.Iterator[dict]:
...
-                ),
+                timeout=TIMEOUT,
+                opt_fields=",".join(USER_FIELDS),
             )
         ]
     finally:
         print(f"Done fetching users for workspace {workspace['name']}.")
 
 
 @dlt.source
```
Comment (rudolfix) [inline pipelines/asana_dlt/__init__.py:270]: IMO @source should be at the top

File: pipelines/asana_dlt/const.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    "followers",
+    "name",
+    "color",
+    "notes",
+    "permalink_url",
+    "workspace",
+]
+
+# List of task fields to be retrieved from Asana API
+TASK_FIELDS = [
```
Comment (rudolfix) [inline pipelines/asana_dlt/const.py:68]: better to use tuples becaue you can make them defaults in the function signatures

File: pipelines/asana_dlt/const.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
+# Default start date for Asana API requests, only tasks started after this date will be collected
+DEFAULT_START_DATE = "2010-01-01T00:00:00.000Z"
+
+# Asana API request timeout
+TIMEOUT = 300
```
Comment (rudolfix) [inline pipelines/asana_dlt/const.py:5]: should be passed as an argument maybe?

Comment (sh-rp) [reply pipelines/asana_dlt/const.py:5]: hm, I would have this as a setting across all endpoints and keep it like this for now

**Pull request merged**